atlasdb-backup/src/test/java/com/palantir/atlasdb/backup/AtlasBackupServiceTest.java

```diff
@@ -85,21 +91,37 @@ public void prepareBackupRegistersLockForRefresh() {
                         authHeader, PrepareBackupRequest.of(ImmutableSet.of(NAMESPACE, OTHER_NAMESPACE))))
                 .thenReturn(PrepareBackupResponse.of(tokens));
 
-        atlasBackupService.prepareBackup(ImmutableSet.of(NAMESPACE, OTHER_NAMESPACE));
+        atlasBackupService.prepareBackup(ImmutableSet.of(ATLAS_SERVICE, OTHER_ATLAS_SERVICE));
         verify(lockRefresher).registerLocks(tokens);
     }
 
     @Test
-    public void completeBackupDoesNotRunUnpreparedNamespaces() {
+    public void prepareBackupThrowsIfNamespacesCollide() {
+        AtlasService collidingAtlasService = AtlasService.of(ServiceId.of("c"), ATLAS_SERVICE.getNamespace());
+        assertThatThrownBy(
+                        () -> atlasBackupService.prepareBackup(ImmutableSet.of(ATLAS_SERVICE, collidingAtlasService)))
+                .isInstanceOf(SafeIllegalArgumentException.class);
```

Comment:
we do have `assertThatLoggableExceptionThrownBy` if you want a more precise assertion here.